### PR TITLE
remove unused functions & types in test/e2e/framework/util.go

### PR DESCRIPTION
**Pull request by andrewsykim**
Signed-off-by: Andrew Sy Kim <kiman@vmware.com>

**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it**:
First pass at cleaning up test/e2e/framework/util.go as part of https://github.com/kubernetes/kubernetes/issues/76206. Longer term we're trying to split up test/e2e/framework/util.go into separate packages for better discovery/readability. This PR is just a first pass to remove any methods/functions/structs that aren't being used. 

**Which issue(s) this PR fixes**:
Part of https://github.com/kubernetes/kubernetes/issues/76206

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:
```release-note
NONE
```

### CODE REVIEW

**Review by neolit123 — COMMENTED** [review]
/lgtm

**Comment by fejta** [comment]
/uncc
Thanks!

**Review by johnSchnake — APPROVED** [review]
👍 Nothing like deleted code :)

If you can delete stuff and it still passes tests, no reason to complain. 💯

File: test/e2e/framework/util.go [hunk 1]
```diff
@@ -224,28 +224,6 @@ var (
...
 )
 
-type Address struct {
-	internalIP string
-	externalIP string
-	hostname   string
-}
-
-// GetServerArchitecture fetches the architecture of the cluster's apiserver.
-func GetServerArchitecture(c clientset.Interface) string {
```
Comment (oomichi) [inline test/e2e/framework/util.go]: no callers as https://cs.k8s.io/?q=GetServerArchitecture&i=nope&files=&repos=kubernetes/kubernetes

File: test/e2e/framework/util.go [hunk 2]
```diff
@@ -362,12 +335,6 @@ func SkipIfMultizone(c clientset.Interface) {
 	}
 }
 
-func SkipUnlessClusterMonitoringModeIs(supportedMonitoring ...string) {
```
Comment (oomichi) [inline test/e2e/framework/util.go]: ditto as https://cs.k8s.io/?q=SkipUnlessClusterMonitoringModeIs&i=nope&files=&repos=kubernetes/kubernetes

File: test/e2e/framework/util.go [hunk 3]
```diff
@@ -386,22 +353,6 @@ func SkipUnlessNodeOSDistroIs(supportedNodeOsDistros ...string) {
 	}
 }
 
-func SkipUnlessSecretExistsAfterWait(c clientset.Interface, name, namespace string, timeout time.Duration) {
```
Comment (oomichi) [inline test/e2e/framework/util.go]: ditto: https://cs.k8s.io/?q=SkipUnlessSecretExistsAfterWait&i=nope&files=&repos=kubernetes/kubernetes

File: test/e2e/framework/util.go [hunk 4]
```diff
@@ -443,15 +394,6 @@ func ProviderIs(providers ...string) bool {
 	return false
 }
 
-func ClusterMonitoringModeIs(monitoringModes ...string) bool {
```
Comment (oomichi) [inline test/e2e/framework/util.go]: SkipUnlessClusterMonitoringModeIs is only caller as https://cs.k8s.io/?q=ClusterMonitoringModeIs&i=nope&files=&repos=kubernetes/kubernetes and that is removed as the above.

File: test/e2e/framework/util.go [hunk 5]
```diff
@@ -586,49 +528,6 @@ func errorBadPodsStates(badPods []v1.Pod, desiredPods int, ns, desiredState stri
 	return errStr + buf.String()
 }
 
-// WaitForPodsSuccess waits till all labels matching the given selector enter
-// the Success state. The caller is expected to only invoke this method once the
-// pods have been created.
-func WaitForPodsSuccess(c clientset.Interface, ns string, successPodLabels map[string]string, timeout time.Duration) error {
```
Comment (oomichi) [inline test/e2e/framework/util.go]: ditto: https://cs.k8s.io/?q=WaitForPodsSuccess&i=nope&files=&repos=kubernetes/kubernetes

File: test/e2e/framework/util.go [hunk 6]
```diff
@@ -804,29 +703,6 @@ func LogFailedContainers(c clientset.Interface, ns string, logFunc func(ftm stri
 	}
 }
 
-func LogPodsWithLabels(c clientset.Interface, ns string, match map[string]string, logFunc func(ftm string, args ...interface{})) {
```
Comment (oomichi) [inline test/e2e/framework/util.go]: WaitForPodsSuccess is only caller as https://cs.k8s.io/?q=LogPodsWithLabels&i=nope&files=&repos=kubernetes/kubernetes and that is removed as the above.

File: test/e2e/framework/util.go [hunk 7]
```diff
@@ -804,29 +703,6 @@ func LogFailedContainers(c clientset.Interface, ns string, logFunc func(ftm stri
...
-		logFunc("Error getting pods in namespace %q: %v", ns, err)
-		return
-	}
-	logFunc("Running kubectl logs on pods with labels %v in %v", match, ns)
-	for _, pod := range podList.Items {
-		kubectlLogPod(c, pod, "", logFunc)
-	}
-}
-
-func LogContainersInPodsWithLabels(c clientset.Interface, ns string, match map[string]string, containerSubstr string, logFunc func(ftm string, args ...interface{})) {
```
Comment (oomichi) [inline test/e2e/framework/util.go]: ditto: https://cs.k8s.io/?q=LogContainersInPodsWithLabels&i=nope&files=&repos=kubernetes/kubernetes

File: test/e2e/framework/util.go [hunk 8]
```diff
@@ -1946,26 +1822,6 @@ func ServerVersionGTE(v *utilversion.Version, c discovery.ServerVersionInterface
 	return sv.AtLeast(v), nil
 }
 
-func SkipUnlessKubectlVersionGTE(v *utilversion.Version) {
```
Comment (oomichi) [inline test/e2e/framework/util.go]: ditto: https://cs.k8s.io/?q=SkipUnlessKubectlVersionGTE&i=nope&files=&repos=kubernetes/kubernetes

File: test/e2e/framework/util.go [hunk 9]
```diff
@@ -1946,26 +1822,6 @@ func ServerVersionGTE(v *utilversion.Version, c discovery.ServerVersionInterface
...
-		Failf("Failed to get kubectl version: %v", err)
-	}
-	if !gte {
-		skipInternalf(1, "Not supported for kubectl versions before %q", v)
-	}
-}
-
-// KubectlVersionGTE returns true if the kubectl version is greater than or
-// equal to v.
-func KubectlVersionGTE(v *utilversion.Version) (bool, error) {
```
Comment (oomichi) [inline test/e2e/framework/util.go]: as https://cs.k8s.io/?q=KubectlVersionGTE&i=nope&files=&repos=kubernetes/kubernetes SkipUnlessKubectlVersionGTE is only caller and that is removed on the above.

File: test/e2e/framework/util.go [hunk 10]
```diff
@@ -3358,40 +3195,6 @@ func isNotRestartAlwaysMirrorPod(p *v1.Pod) bool {
 	return p.Spec.RestartPolicy != v1.RestartPolicyAlways
 }
 
-// Waits for the number of events on the given object to reach a desired count.
-func WaitForEvents(c clientset.Interface, ns string, objOrRef runtime.Object, desiredEventsCount int) error {
```
Comment (oomichi) [inline test/e2e/framework/util.go]: ditto: https://cs.k8s.io/?q=WaitForEvents&i=nope&files=&repos=kubernetes/kubernetes

File: test/e2e/framework/util.go [hunk 11]
```diff
@@ -3358,40 +3195,6 @@ func isNotRestartAlwaysMirrorPod(p *v1.Pod) bool {
...
-		if eventsCount < desiredEventsCount {
-			return false, nil
-		}
-		// Number of events has exceeded the desired count.
-		return false, fmt.Errorf("number of events has exceeded the desired count, eventsCount: %d, desiredCount: %d", eventsCount, desiredEventsCount)
-	})
-}
-
-// Waits for the number of events on the given object to be at least a desired count.
-func WaitForPartialEvents(c clientset.Interface, ns string, objOrRef runtime.Object, atLeastEventsCount int) error {
```
Comment (oomichi) [inline test/e2e/framework/util.go]: ditto: https://cs.k8s.io/?q=WaitForPartialEvents&i=nope&files=&repos=kubernetes/kubernetes

File: test/e2e/framework/util.go [hunk 12]
```diff
@@ -4538,41 +4341,6 @@ func getKubeletPods(c clientset.Interface, node, resource string) (*v1.PodList,
 	return result, nil
 }
 
-// LaunchWebserverPod launches a pod serving http on port 8080 to act
-// as the target for networking connectivity checks.  The ip address
-// of the created pod will be returned if the pod is launched
-// successfully.
-func LaunchWebserverPod(f *Framework, podName, nodeName string) (ip string) {
```
Comment (oomichi) [inline test/e2e/framework/util.go]: ditto: https://cs.k8s.io/?q=LaunchWebserverPod&i=nope&files=&repos=kubernetes/kubernetes

File: test/e2e/framework/util.go [hunk 13]
```diff
@@ -4660,24 +4428,6 @@ func parseSystemdServices(services string) string {
 	return strings.TrimSpace(strings.Replace(services, ",", " ", -1))
 }
 
-func UpdatePodWithRetries(client clientset.Interface, ns, name string, update func(*v1.Pod)) (*v1.Pod, error) {
```
Comment (oomichi) [inline test/e2e/framework/util.go]: ditto: https://cs.k8s.io/?q=UpdatePodWithRetries&i=nope&files=&repos=kubernetes/kubernetes

**Review by oomichi — COMMENTED** [review]
confirmed everything is fine for me, thanks.

/lgtm

**Comment by andrewsykim** [comment]
Thanks for the thorough check @oomichi!!

**Pull request merged**